core/src/main/java/com/datastrato/gravitino/proto/RoleEntitySerDe.java

```diff
@@ -55,31 +63,39 @@ public Role serialize(RoleEntity roleEntity) {
    */
   @Override
   public RoleEntity deserialize(Role role, Namespace namespace) {
-    List<Privilege> privileges = Lists.newArrayList();
-
-    if (!role.getPrivilegesList().isEmpty()) {
+    List<SecurableObject> securableObjects = Lists.newArrayList();
 
-      for (int index = 0; index < role.getPrivilegeConditionsCount(); index++) {
-        if (Privilege.Condition.ALLOW.name().equals(role.getPrivilegeConditions(index))) {
-          privileges.add(Privileges.allow(role.getPrivileges(index)));
+    for (int index = 0; index < role.getSecurableObjectsCount(); index++) {
+      List<Privilege> privileges = Lists.newArrayList();
+      for (int privIndex = 0;
+          privIndex < role.getSecurableObjects(index).getPrivilegeConditionsCount();
+          privIndex++) {
+        if (Privilege.Condition.ALLOW
+            .name()
+            .equals(role.getSecurableObjects(index).getPrivilegeConditions(privIndex))) {
+          privileges.add(
+              Privileges.allow(role.getSecurableObjects(index).getPrivilegeNames(privIndex)));
```

Comment:
Can you please add more tests to verify the serialization and deserialization of this part?

Reply:
Done.